### DISCUSSION TITLE
Rows: Repeating rows does not work with shared query / dashboard data source

### DISCUSSION THREAD

**Discussion by rfriedma** [grafana/grafana#38182] [Dashboards, Panels & Visualizations]
**What happened**: 
I set up a row to be repeated on my dashboard. For the row definition, I set it up such that there was a primary query utilized by the first panel in the row. The second panel was set up such that it used the data from the first panel in the row and just visualized a different column in the tabular data returned by the primary query. When I repeated the row by using a dashboard variable, the second row's second tile referenced the first row's data instead of the second.
![image](https://user-images.githubusercontent.com/41642157/100641402-0a8c1780-32fd-11eb-9943-0f49e0a3d462.png)

**What you expected to happen**:
I expected the second row's second panel to reference the second row's primary panel.
![image](https://user-images.githubusercontent.com/41642157/100643167-517b0c80-32ff-11eb-9c59-e2926d26133b.png)

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a dashboard with a variable that can be used for a data query and used for repeating a row
2. Create a Row
3. Create a Panel in the row and use the variable in the query for that panel
4. Create a second Panel in the row and use the first Panel's data source ("--Dashboard--" option).
5. Set the Row to repeat on the variable
6. Select more than one variable value such that the Row repeats

**Anything else we need to know?**:

**Environment**:
- Grafana version: 7.2.2
- Data source type & version: Custom data source returning tabular data
- OS Grafana is installed on: Windows Server 2016 64-bit
- User OS & Browser: Windows 10 Enterprise and Chrome 87.0.4280.66 (Official Build) (64-bit)
- Grafana plugins: Custom graph panel to support tabular data
- Others:

**Comment by torkelo** [comment]
this is currently not supported so marked this as feature request as it could require some big changes to support

**Comment by rfriedma** [comment]
It honestly feels like broken behavior when combining shared queries and repeated rows. Individually they work, together they don't make sense.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by jdraymon** [comment]
Worth at least adding some documentation? Was kind of a "gotcha" for me when the values in the repeated rows were all wrong (using vanilla plugins - influx, singlestat, etc)

**Comment by griseau** [comment]
Hello guys, 
Do you plan to support it or not at all ? 

It could improve loading time of dashboard very easily ! 

Thanks

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82966. Feel free to continue the discussion around this there. Thank you!
